pubnub/lib/src/crypto/crypto.dart

```diff
@@ -1,132 +1,106 @@
-import 'package:encrypt/encrypt.dart' as crypto;
-import 'package:crypto/crypto.dart' show sha256;
-import 'dart:convert' show base64, utf8;
-import 'dart:typed_data' show Uint8List;
-
 import 'package:pubnub/core.dart';
-import 'encryption_mode.dart';
-
-/// Configuration used in cryptography.
-class CryptoConfiguration {
-  /// Encryption mode used.
-  final EncryptionMode encryptionMode;
-
-  /// Whether key should be encrypted.
-  final bool encryptKey;
-
-  /// Whether a random IV should be used.
-  final bool useRandomInitializationVector;
+import 'package:pubnub/src/crypto/aesCbcCryptor.dart';
+import 'package:pubnub/src/crypto/legacyCryptor.dart';
+import 'cryptoConfiguration.dart';
+import 'cryptorHeader.dart';
 
-  const CryptoConfiguration(
-      {this.encryptionMode = EncryptionMode.CBC,
-      this.encryptKey = true,
-      this.useRandomInitializationVector = true});
-}
-
-/// Default cryptography module used in PubNub SDK.
+/// CryptoModule is responsible for encryption and decryption
+/// of PubNub messages.
 class CryptoModule implements ICryptoModule {
-  final CryptoConfiguration defaultConfiguration;
+  final ICryptor defaultCryptor;
+  List<ICryptor>? cryptors;
+
+  late CryptoConfiguration defaultConfiguration;
+  late LegacyCryptoModule legacyCryptoModule;
+
+  CryptoModule(
+      {required this.defaultCryptor,
+      this.cryptors,
+      this.defaultConfiguration = const CryptoConfiguration()}) {
+    legacyCryptoModule =
+        LegacyCryptoModule(defaultConfiguration: defaultConfiguration);
+  }
 
-  /// Default configuration is:
-  /// * `encryptionMode` set to [EncryptionMode.CBC].
-  /// * `encryptKey` set to `true`.
-  /// * `useRandomInitializationVector` set to `true`.
-  CryptoModule({this.defaultConfiguration = const CryptoConfiguration()});
+  factory CryptoModule.legacyCryptoModule(CipherKey cipherKey,
+      {defaultCryptoConfiguration = const CryptoConfiguration()}) {
+    return CryptoModule(
+        defaultCryptor: LegacyCryptor(cipherKey,
+            cryptoConfiguration: defaultCryptoConfiguration),
+        cryptors: <ICryptor>[AesCbcCryptor(cipherKey)]);
+  }
 
-  crypto.Key _getKey(CipherKey cipherKey, CryptoConfiguration configuration) {
-    if (configuration.encryptKey) {
-      return crypto.Key.fromUtf8(
-          sha256.convert(cipherKey.data).toString().substring(0, 32));
-    } else {
-      return crypto.Key(Uint8List.fromList(cipherKey.data));
-    }
+  factory CryptoModule.aescbcCryptoModule(CipherKey cipherKey,
```

Comment:
Nitpick: typo in method's name (could be camel-cased `aesCbcCryptoModule`)

Reply:
yeah! naming!